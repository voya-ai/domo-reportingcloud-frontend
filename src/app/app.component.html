<div class="container-fluid">
  <div class="row">
    <nav class="col-sm-3 col-md-2 d-none d-sm-block bg-light sidebar">
      <ul class="nav nav-pills flex-column">
        <li class="nav-item">
          <a routerLink="/"
              routerLinkActive="active"
              class="nav-link">View reports</a>
        </li>
        <li class="nav-item">
          <a routerLink="/report/add"
              routerLinkActive="active"
              class="nav-link">Add report</a>
        </li>
      </ul>
    </nav>
  </div>
</div>

<main class="col-sm-9 ml-sm-auto col-md-10 pt-3" role="main">
  <router-outlet></router-outlet>
</main>
