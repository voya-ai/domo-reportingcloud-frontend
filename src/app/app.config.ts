import { OpaqueToken } from "@angular/core";
import { environment } from 'environments/environment';

export let APP_CONFIG = new OpaqueToken("app.config");

export interface IAppConfig {
    apiEndpoint: string;
}

export const AppConfig: IAppConfig = {
    apiEndpoint: environment.apiUrl
};
