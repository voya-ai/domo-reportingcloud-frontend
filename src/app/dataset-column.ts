export interface DatasetColumn {
  name: string,
  checked: boolean
}
