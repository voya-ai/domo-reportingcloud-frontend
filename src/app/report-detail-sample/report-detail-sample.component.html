<h2>Sample</h2>

<div>
  <div class="form-group">
    <label>Child group (different documents)</label>
    <mat-checkbox *ngFor="let column of childGroupColumns" [(ngModel)]="column.childGroup" class="form-check">
      {{column.name}}
    </mat-checkbox>
  </div>
</div>

<div>
  <div *ngIf="!loading; else showSpinner">
    <button
      type="button"
      class="btn btn-primary"
      (click)="sample(report)">
      Download sample
    </button>
  </div>
  <ng-template #showSpinner>
    <mat-spinner></mat-spinner>
  </ng-template>
</div>

<div class="form-group" *ngIf="data">
  <label>Data</label>
  <textarea class="form-control" rows="5">
    {{data}}
  </textarea>
</div>
