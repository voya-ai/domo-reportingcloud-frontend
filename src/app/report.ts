export interface Report {
  name?: string;
  slug: string;
  dataSet?: string;
  templateName?: string;
  headerColumns?: string[];
  contentColumns?: string[];
}
