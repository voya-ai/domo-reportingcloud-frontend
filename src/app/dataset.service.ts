import { Injectable, Inject } from '@angular/core';
import { Http } from '@angular/http';
import { Observable } from 'rxjs/Observable';
import { APP_CONFIG, IAppConfig } from './app.config';
import { DatasetColumn } from './dataset-column';

import 'rxjs/add/operator/map';
import 'rxjs/add/operator/filter';
import 'rxjs/add/operator/first';
import 'rxjs/add/operator/shareReplay';

@Injectable()
export class DatasetService {
  private data : Observable<any[]>;

  constructor(
    private _http: Http,
    @Inject(APP_CONFIG) private config
  ) {
    this.data = this._http.get(`${this.config.apiEndpoint}/datasets`)
      .map(res => res.json())
      .shareReplay(1);
  }

  search(term: string) : Observable<any[]> {
    return this.data
      .map
      (data => {
        return data.filter(item => {
          return item.name.search(new RegExp(term, "i")) !== -1;
        });
      });
  }

  get(id: string) : Observable<any> {
    return this._http.get(`${this.config.apiEndpoint}/datasets/${id}`)
      .map(res => res.json());
  }

  getColumns(id: string) : Observable<DatasetColumn[]> {
    return this.get(id)
      .map(res => res.schema.columns.map(column => {
        return {
          name: column.name,
          headerChecked: false,
          contentChecked: false
        }
      })).map(data => {
        return data.sort((a, b) => {
          if(a.name < b.name) return -1;
          if(a.name > b.name) return 1;
          return 0;
        });
      });
  }

}
