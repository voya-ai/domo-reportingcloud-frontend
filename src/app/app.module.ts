import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { HttpModule } from '@angular/http';

import { RouterModule, Routes } from '@angular/router';

import {
  MatAutocompleteModule,
  MatInputModule,
  MatCardModule,
  MatGridListModule,
  MatCheckboxModule,
  MatButtonModule,
  MatDatepickerModule,
  MatNativeDateModule,
  MatSelectModule,
  MatMenuModule,
  MatListModule,
  MatToolbarModule,
  MatProgressSpinnerModule
} from '@angular/material';

import { AppComponent } from './app.component';

import { APP_CONFIG, AppConfig } from './app.config';

import { ReportService } from './report.service';
import { DatasetService } from './dataset.service';
import { TemplateService } from './template.service';
import { ReportGenerationService } from './report-generation.service';

import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ReportListComponent } from './report-list/report-list.component';
import { ReportAddComponent } from './report-add/report-add.component';
import { ReportDetailComponent } from './report-detail/report-detail.component';
import { ReportDatasetAutocompleteComponent } from './report-dataset-autocomplete/report-dataset-autocomplete.component';
import { ReportTemplateAutocompleteComponent } from './report-template-autocomplete/report-template-autocomplete.component';
import { ReportGenerationListComponent } from './report-generation-list/report-generation-list.component';
import { ReportsComponent } from './reports/reports.component';
import { ReportDetailSampleComponent } from './report-detail-sample/report-detail-sample.component';
import { ReportEditComponent } from './report-edit/report-edit.component';
import { DatasetColumnSelectComponent } from './dataset-column-select/dataset-column-select.component';

const appRoutes: Routes = [
    {
        path: 'report/add',
        component: ReportAddComponent
    },
    {
        path: 'report/:slug',
        component: ReportDetailComponent
    },
    {
        path: '',
        component: ReportsComponent,
        pathMatch: 'full'
    }
];

@NgModule({
  declarations: [
    AppComponent,
    ReportListComponent,
    ReportAddComponent,
    ReportDetailComponent,
    ReportDatasetAutocompleteComponent,
    ReportTemplateAutocompleteComponent,
    ReportGenerationListComponent,
    ReportsComponent,
    ReportDetailSampleComponent,
    ReportEditComponent,
    DatasetColumnSelectComponent
  ],
  imports: [
    BrowserModule,
    FormsModule,
    HttpModule,
    ReactiveFormsModule,
    NoopAnimationsModule,
    RouterModule.forRoot(appRoutes),
    MatAutocompleteModule,
    MatInputModule,
    MatCardModule,
    MatGridListModule,
    MatCheckboxModule,
    MatButtonModule,
    MatDatepickerModule,
    MatNativeDateModule,
    MatSelectModule,
    MatMenuModule,
    MatListModule,
    MatToolbarModule,
    MatProgressSpinnerModule
  ],
  providers: [
      { provide: APP_CONFIG, useValue: AppConfig },
      ReportService,
      DatasetService,
      TemplateService,
      ReportGenerationService
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
