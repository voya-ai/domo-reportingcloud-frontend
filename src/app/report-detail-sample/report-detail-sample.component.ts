import { Component, Input, OnInit } from '@angular/core';

import { ReportService } from 'app/report.service';

import { Report } from 'app/report';

@Component({
  selector: 'app-report-detail-sample',
  templateUrl: './report-detail-sample.component.html',
  styleUrls: ['./report-detail-sample.component.css']
})
export class ReportDetailSampleComponent implements OnInit {
  @Input() report: Report;

  childGroupColumns: any[];
  loading: boolean = false;
  data: string;

  constructor(private _reportService: ReportService) { }

  ngOnInit() {
    this.childGroupColumns = this.report.headerColumns.map((column) => {
      return { name: column, childGroup: false }
    });
  }

  sample(report: Report) {
    this.loading = true;
    this.data = null;

    this._reportService.sample(report, {
      childGroups: this.childGroupColumns.filter(column => column.childGroup).map(column => column.name)
    }).subscribe(data => {
      this.loading = false;
      this.data = data;
    });
  }
}
