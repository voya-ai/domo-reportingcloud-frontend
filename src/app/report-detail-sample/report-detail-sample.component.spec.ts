import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { Observable } from 'rxjs/Observable';
import {
  MatCheckboxModule
} from '@angular/material';

import { Report } from 'app/report';
import { ReportService } from 'app/report.service';

import 'rxjs/add/observable/of';

import { ReportDetailSampleComponent } from './report-detail-sample.component';

class MockReportService {}

describe('ReportDetailSampleComponent', () => {
  let component: ReportDetailSampleComponent;
  let fixture: ComponentFixture<ReportDetailSampleComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      providers: [
        { provide: ReportService, useClass: MockReportService },
      ],
      declarations: [ ReportDetailSampleComponent ],
      imports: [
        MatCheckboxModule,
        FormsModule,
        ReactiveFormsModule
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ReportDetailSampleComponent);
    component = fixture.componentInstance;
    component.report = {
      name: "Test",
      slug: "test",
      headerColumns: []

    }
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
