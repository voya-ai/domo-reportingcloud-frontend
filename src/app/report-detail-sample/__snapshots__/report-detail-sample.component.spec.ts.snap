// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDetailSampleComponent should match the snapshot 1`] = `
<app-report-detail-sample
  _reportService={[Function MockReportService]}
  childGroupColumns={[Function Array]}
  loading="false"
  report={[Function Object]}
>
  <h2>
    Sample
  </h2>
  
  <div>
    
    
    <div
      class="form-group"
    >
      
      
      <label>
        Child group (different documents)
      </label>
      
      
      
      
    
    </div>
    
  
  </div>
  
  <div>
    
    
    
    <div>
      
      
      <button
        class="btn btn-primary"
        type="button"
      >
        
        Download sample
      
      </button>
      
    
    </div>
    
    
    
    
  
  </div>
  
  
  
</app-report-detail-sample>
`;
